apps/design/backend/src/worker/generate_election_and_ballot_packages.ts

```diff
@@ -142,33 +157,89 @@ export async function generateElectionPackage(
     JSON.stringify(uiStringAudioIds, null, 2)
   );
-  zip.file(ElectionPackageFileName.AUDIO_CLIPS, uiStringAudioClips);
+  electionPackageZip.file(
+    ElectionPackageFileName.AUDIO_CLIPS,
+    uiStringAudioClips
+  );
 
   if (ballotTemplateId === 'NhBallotV3') {
-    makeV3Compatible(zip, systemSettings);
+    makeV3Compatible(electionPackageZip, systemSettings);
   }
 
-  const zipContents = await zip.generateAsync({
+  const electionPackageZipContents = await electionPackageZip.generateAsync({
     type: 'nodebuffer',
     streamFiles: true,
   });
-  const electionPackageHash = sha256(zipContents);
+  // No need to hash the ballot pdfs since they are entirely derived from election package data
+  const electionPackageHash = sha256(electionPackageZipContents);
+
   const combinedHash = formatElectionHashes(
     electionDefinition.ballotHash,
     electionPackageHash
   );
-  const fileName = `election-package-${combinedHash}.zip`;
+  const electionPackageFileName = `election-package-${combinedHash}.zip`;
+
+  electionPackageZip.file(
+    ElectionPackageFileName.APP_STRINGS,
+    JSON.stringify(appStrings, null, 2)
+  );
+
+  combinedZip.file(electionPackageFileName, electionPackageZipContents);
+
+  // Make ballot package zip
+  for (const [props, document] of iter(allBallotProps).zip(ballotDocuments)) {
+    const pdf = await document.renderToPdf();
+    const { precinctId, ballotStyleId, ballotType, ballotMode } = props;
+    const precinct = assertDefined(getPrecinctById({ election, precinctId }));
+    const fileName = getPdfFileName(
+      precinct.name,
+      ballotStyleId,
+      ballotType,
+      ballotMode
+    );
+    ballotPackageZip.file(fileName, pdf);
+  }
+
+  const readinessReportPdf = await renderBallotStyleReadinessReport({
+    componentProps: {
+      electionDefinition,
+      generatedAtTime: new Date(),
+    },
+    renderer,
+  });
+
+  // eslint-disable-next-line no-console
+  renderer.cleanup().catch(console.error);
+
+  ballotPackageZip.file(
+    BALLOT_STYLE_READINESS_REPORT_FILE_NAME,
+    readinessReportPdf
+  );
+
+  // Add ballot package to combined zip
+  const ballotPackageZipContents = await ballotPackageZip.generateAsync({
+    type: 'nodebuffer',
+    streamFiles: true,
+  });
+  const ballotPackageFileName = `ballot-package-${combinedHash}.zip`;
+  combinedZip.file(ballotPackageFileName, ballotPackageZipContents);
 
+  // Write combined zip to file storage
+  const combinedFileName = `combined-election-and-ballot-package-${combinedHash}.zip`;
+  const combinedZipContents = await combinedZip.generateAsync({
+    type: 'nodebuffer',
+    streamFiles: true,
+  });
   const writeResult = await fileStorageClient.writeFile(
-    path.join(orgId, fileName),
-    zipContents
+    path.join(orgId, combinedFileName),
+    combinedZipContents
   );
   writeResult.unsafeUnwrap();
-  const electionPackageUrl = `/files/${orgId}/${fileName}`;
+  const electionPackageUrl = `/files/${orgId}/${combinedFileName}`;
```

Comment:
This all looks great! Just thought I'd drop some quick 2 cents on naming conventions here. I think I lean toward something like:

> `election-package-and-ballots-${combinedHash}.zip`
> --> `ballots-${ballotHash}.zip`
> --> `election-package-${combinedHash}.zip`

Main edits are:
- Dropping combined from the combined name as it's implied, and saying `election-package-and-ballots` rather than `election-and-ballot-package` since `ballot-package` could be confusing as it's what we used to call the `election-package`
- Only using the ballot hash and not the combined hash in the name of the ballots ZIP since the second part of the combined hash isn't relevant to it

Reply:
Good point, updated

![Screenshot 2025-02-04 at 9 11 59 AM](https://github.com/user-attachments/assets/53b28587-a245-4eab-8c5b-c15b57375837)